pkg/kubecli/vm.go

```diff
@@ -273,6 +299,57 @@ func (v *vms) subresourceHelper(name string, resource string, in io.Reader, out
 		return fmt.Errorf("no response received")
 	}
 }
+func (v *vms) asyncSubresourceHelper(name string, resource string, done chan struct{}) (*websocket.Conn, error) {
+
+	obj := &asyncWSCallbackObj{
+		Connection: make(chan *websocket.Conn),
+		Done:       done,
+	}
+	// Create a round tripper with all necessary kubernetes security details
+	wrappedRoundTripper, err := roundTripperFromConfig(v.config, obj.WebsocketCallback)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create round tripper for remote execution: %v", err)
+	}
+
+	// Create a request out of config and the query parameters
+	req, err := RequestFromConfig(v.config, name, v.namespace, resource)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create request for remote execution: %v", err)
+	}
+
+	errChan := make(chan error)
+	go func() {
+		// Send the request and let the callback do its work
+		response, err := wrappedRoundTripper.RoundTrip(req)
+
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		if response != nil {
+			switch response.StatusCode {
+			case http.StatusOK:
+			case http.StatusNotFound:
+				err = fmt.Errorf("Virtual Machine not found.")
+			case http.StatusInternalServerError:
+				err = fmt.Errorf("Websocket failed due to internal server error.")
+			default:
+				err = fmt.Errorf("Websocket failed with http status: %s", response.Status)
+			}
+		} else {
+			err = fmt.Errorf("no response received")
+		}
+		errChan <- err
+	}()
+
+	select {
+	case err = <-errChan:
```

Comment:
@fromanirh related to #934. This is similar what we want to do in general.

Reply:
@rmohr got it